pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -96,7 +96,13 @@ func GetValidVolumeSources(clientset kubernetes.Interface, namespace string, vol
 		if err != nil {
 			return validVolumes, err
 		}
-		if len(osdDeployments.Items) == 0 && len(osdPrepareJob.Items) == 0 {
+		if len(osdDeployments.Items) == 0 {
+			if len(osdPrepareJob.Items) == 0 || len(osdPrepareJob.Items) == 1 {
+				validVolumes = append(validVolumes, vol)
+			}
+		}
+
+		if isUpgrade {
```

Comment:
Why do we even need this method for `GetValidVolumeSources()`? The normal flow for OSDs during orchestration should always be:
- Run the OSD prepare job. Inside the prepare job if the OSD has already been created, c-v should detect it already exists on that device and then return. 
- Create the OSD daemon deployment. If it already exists, update the spec. (this is needed for upgrades)

I can't remember now why we wanted to skip the prepare job completely if it had already been run previously. Maybe that's an old assumption that we can fix now? Or maybe you've already fixed it with the change to skip releasing the LV?

Reply:
In any case, looks like the volume could be added to the slice twice under some conditions, on lines 101 and 106.

Reply:
Since we're skipping c-v lvm prepare for the already existing OSDs, can't we always call the osd prepare job? We shouldn't only do this for upgrades, but we should do this on every orchestration in that case. I don't see why we need to check for `isUpgrade`. I'm saying we should follow the pattern we have for other osds that will call the osd prepare job with every orchestration even if it's a no-op.

Reply:
`GetValidVolumeSources()` has been removed now. We are now always creating the OSD prepare job  on operator reconcile and skipping `lvm prepare` if OSD is already created on the PVC device.